### PR TITLE
Configurator default keymaps

### PR DESCRIPTION
**Pull request by s-show**
## Description

Translate `docs/configurator_default_keymaps.md` to Japanese.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/ja/configurator_default_keymaps.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  original document: 0.13.15:docs/configurator_default_keymaps.md
+  git diff 0.13.15 HEAD -- docs/configurator_default_keymaps.md | cat
+-->
+
+このページは、QMK Configurator にキーボードのデフォルトキーマップを追加する方法を取り上げます。
+
+
+## 技術情報 :id=technical-information
+
+QMK Configurator は、キーマップのネイティブファイル形式に JSON ファイルを使います。可能な限り、QMK Configurator で作成したファームウェアと、`qmk_firmware` で `make <keyboard>:default` を実行して作成したファームウェアが同じ動作となるべきです。
```
Comment (s-show) [inline docs/ja/configurator_default_keymaps.md:12]: ここの翻訳に当たっては、原文の著者に内容を質問して回答を得ています。その応答は #10049 のコメント欄の末尾で見られます。

**Comment by mtei** [comment]
@umi-umi @shelaf @hsgw @satt99 @takashiski @yoshihara よかったらレビューお願いします

File: docs/ja/configurator_default_keymaps.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+enum layer_names {
+    _BASE,
+    _MEDIA,
+    _FN
+};
+```
+
+これは C 言語では動作しますが、QMK Configurator では動作しません。レイヤーのインデックスには数値を使用する *必要があります*。上記の例だと、`MO(_FN)` ではなく `MO(2)` とする必要があります。
+
+### いくつかのカスタムコードはサポート対象外 :id=custom-code
```
Comment (shelaf) [inline docs/ja/configurator_default_keymaps.md:119]: any kindなので例外はなし
```suggestion
### カスタムコードはサポート対象外 :id=custom-code
```

File: docs/ja/configurator_default_keymaps.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+            if (record->event.pressed) {
+                SEND_STRING("This is macro #3.");
+            }
+            return false;
+    }
+    return true;
+};
+```
+
+... `enum` ブロックのキーコードをキーボードのヘッダーファイル (`<keyboard>.h`) に次のとおり追加します。 (`enum` がここでは `keyboard_keycodes` と命名されていることに注意してください):
```
Comment (shelaf) [inline docs/ja/configurator_default_keymaps.md:157]:
```suggestion
... `enum` ブロックのキーコードをキーボードのヘッダーファイル (`<keyboard>.h`) に次のように追加します。 (`enum` がここでは `keyboard_keycodes` と命名されていることに注意してください):
```

File: docs/ja/configurator_default_keymaps.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+            if (record->event.pressed) {
+                SEND_STRING("This is macro #3.");
+            }
+            return false;
+    }
+    return process_record_user(keycode, record);
+};
+```
+
+最後に `process_record_user()` を呼び出すことに注意してください。さらに、ユーザーが自身のカスタムキーコードをキーボードによって提供される以上にキーマップレベルで追加したい場合のために、`SAFE_RANGE` の代わりに `NEW_SAFE_RANGE` を使う必要があります。
```
Comment (shelaf) [inline docs/ja/configurator_default_keymaps.md:193]:
```suggestion
最後に `process_record_user()` を呼び出すことに注意してください。さらに、ユーザーが自身のカスタムキーコードをキーボードによって提供される以上にキーマップレベルで追加したい場合は、`SAFE_RANGE` の代わりに `NEW_SAFE_RANGE` を使う必要があります。
```

**Review by shelaf — APPROVED** [review]

File: docs/ja/configurator_default_keymaps.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+* `keymap` (文字列)
+  * `default` と設定する必要があります。
+* `layout` (文字列)
+  * デフォルトキーマップによって使われるレイアウトマクロです。
+* `layers` (配列)
+  * キーマップそのものです。このキーは、レイヤー毎に1つの配列を含む必要があり、配列はレイヤーを構成するキーコードを含む必要があります。
+
+さらに、たいていのキーマップは `commit` キーを含みます。このキーは、QMK Configurator を支える API では使用されませんが、QMK Configurator のメンテナーがリポジトリ内の JSON 形式のキーマップを作るために使われたキーマップのバージョンを通知するために使われます。この値は、`qmk_firmware` リポジトリ内のキーボードのデフォルトの `keymap.c` を変更する最後のコミットの SHA です。この SHA の値は、[`qmk_firmware` リポジトリの `master` ブランチ](https://github.com/qmk/qmk_firmware/tree/master/) をチェックアウトして、`git log -1 --pretty=oneline -- keyboards/<keyboard>/keymaps/default/keymap.c` (問題のキーボードに `keymap.json` ファイルがあれば、代わりにそれを使います。) を実行すると、次のようなものが返ってきます:
+
+```shell
```
Comment (peepeetee) [inline docs/ja/configurator_default_keymaps.md:27]: There's no longer a "shell" on this line

File: docs/ja/configurator_default_keymaps.md [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    return process_record_user(keycode, record);
+};
+```
+
+最後に `process_record_user()` を呼び出すことに注意してください。さらに、ユーザーが自身のカスタムキーコードをキーボードによって提供される以上にキーマップレベルで追加したい場合は、`SAFE_RANGE` の代わりに `NEW_SAFE_RANGE` を使う必要があります。
+
+
+## 追加説明 :id=additional-reading
+
+QMK Configurator があなたのキーボードをサポートするためには、あなたのキーボードが `qmk_firmware` リポジトリの `master` ブランチに存在していなければなりません。この点の説明は、[キーボードサポート](https://docs.qmk.fm/#/ja/reference_configurator_support)を参照してください。
```
Comment (peepeetee) [inline docs/ja/configurator_default_keymaps.md:198]:
```suggestion
QMK Configurator があなたのキーボードをサポートするためには、あなたのキーボードが `qmk_firmware` リポジトリの `master` ブランチに存在していなければなりません。この点の説明は、[キーボードサポート](ja/reference_configurator_support)を参照してください。
```

File: docs/ja/configurator_default_keymaps.md (hunk 5 above)
Comment (peepeetee) [reply docs/ja/configurator_default_keymaps.md:27]: line 27

**Comment by zvecr** [comment]
Closing as this would have gone stale multiple times by now.

**Pull request closed without merging**